File: CS/CustomSavingEUD/OpenCommandHandler.cs
using System.Windows.Forms;
using DevExpress.XtraReports.UserDesigner;

public class OpenCommandHandler : ICommandHandler {
    public OpenCommandHandler() {
    }

    public void HandleCommand(ReportCommand command,
    object[] args) {
        switch (command) {
            case ReportCommand.OpenFile:
                Open();
                break;
        }
    }

    public bool CanHandleCommand(ReportCommand command,
    ref bool useNextHandler) {
        useNextHandler = !(command == ReportCommand.OpenFile);
        return !useNextHandler;
    }

    void Open() {
        MessageBox.Show("Open");
    }
}

File: VB/CustomSavingEUD/OpenCommandHandler.vb
Imports DevExpress.XtraReports.UserDesigner

Public Class OpenCommandHandler
	Implements ICommandHandler

	Public Sub New()
	End Sub

	Public Sub HandleCommand(ByVal command As ReportCommand, ByVal args() As Object) Implements ICommandHandler.HandleCommand
		Select Case command
			Case ReportCommand.OpenFile
				Open()
		End Select
	End Sub

	Public Function CanHandleCommand(ByVal command As ReportCommand, ByRef useNextHandler As Boolean) As Boolean Implements ICommandHandler.CanHandleCommand
		useNextHandler = Not (command = ReportCommand.OpenFile)
		Return Not useNextHandler
	End Function

	Private Sub Open()
		MessageBox.Show("Open")
	End Sub
End Class
